doc/website/blog/2023-12-12-hibernate-schema-loader.mdx

```diff
@@ -0,0 +1,591 @@
+
+If you haven't already, install the latest version of Atlas:
+
+<InstallationInstructions />
+
+Create a project and add the plugin and the following dependencies:
+
+```gradle
+plugins {
+    id("io.atlasgo.hibernate-provider-gradle-plugin") version "0.1"
+}
+
+dependencies {
+    implementation("org.hibernate.orm:hibernate-core:6.4.0.Final")
+    implementation("org.postgresql:postgresql:42.7.0")
+}
+```
+
+Copy the following Java files to the directory `src/main/java/org/example`:
+
+<Tabs>
+<TabItem value="Actor">
+
+```java title=src/main/java/org/example/Actor.java
+package org.example;
+
+import jakarta.persistence.*;
+
+import java.util.List;
+import java.util.Set;
+
+@Entity
+@Table(name = "Actors")
+public class Actor {
+    Actor() {
+
+    }
+
+    Actor(String name) {
+        this.name = name;
+    }
+
+    @Id
+    public String name;
+
+    @OneToMany(mappedBy = "actor", cascade = CascadeType.PERSIST)
+    public Set<MovieParticipation> actedIn;
+}
+```
+
+</TabItem>
+<TabItem value="Movie">
+
+```java title=src/main/java/org/example/Movie.java
+package org.example;
+
+import jakarta.persistence.*;
+
+import java.util.List;
+
+@Entity
+@Table(name = "Movies")
+public class Movie {
+    Movie(String title, Integer numberInSeries) {
+        this.title = title;
+        this.numberInSeries = numberInSeries;
+    }
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    public Long id;
+
+    public String title;
+
+    public Integer numberInSeries;
+}
+
+```
+
+</TabItem>
+<TabItem value="MovieParticipation">
+
+```java title=src/main/java/org/example/MovieParticipation.java
+package org.example;
+
+import jakarta.persistence.*;
+
+@Entity
+public class MovieParticipation {
+    MovieParticipation(Movie movie, Actor actor) {
+        this.key = new MovieParticipationKey(movie.id, actor.name);
+        this.actor = actor;
+        this.movie = movie;
+    }
+
+    @EmbeddedId
+    public MovieParticipationKey key;
+
+    @ManyToOne(cascade=CascadeType.PERSIST)
+    @MapsId("movieId")
+    @JoinColumn(name = "movieId")
+    Movie movie;
+
+    @ManyToOne(cascade=CascadeType.PERSIST)
+    @MapsId("actorName")
+    @JoinColumn(name = "actorName")
+    Actor actor;
+}
+```
+
+</TabItem>
+<TabItem value="MovieParticipationKey">
+
+```java title=src/main/java/org/example/MovieParticipationKey.java
+package org.example;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Embeddable;
+
+import java.io.Serializable;
+
+@Embeddable
+public class MovieParticipationKey implements Serializable {
+    public MovieParticipationKey(Long movieId, String actorName) {
+        this.movieId = movieId;
+        this.actorName = actorName;
+    }
+    public Long movieId;
+
+    public String actorName;
+}
+```
+
+</TabItem>
+<TabItem value="Main">
+
+```java title=src/main/java/org/example/Main.java
```

Comment:
We can keep the article more concise by focusing only on the migrations part. The e2e is example for testing internally but doesn't add anything to the story. So remove this part